compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -290,50 +290,56 @@ object Types {
      *  instance, or NoSymbol if none exists (either because this type is not a
      *  value type, or because superclasses are ambiguous).
      */
-    final def classSymbol(implicit ctx: Context): Symbol = this match {
-      case ConstantType(constant) =>
-        constant.tpe.classSymbol
```

Comment:
wasn't this already `tailrec` optimized?
Instead of rewriting by hand I suggest using https://github.com/lampepfl/dotty/issues/1221